### DISCUSSION TITLE
Multi-app Monorepo Followup

### DISCUSSION THREAD

**Discussion by mysterycommand** [npm/feedback#582] [CLI] [answered]
Following up on https://github.com/npm/rfcs/issues/463 and https://github.com/npm/rfcs/issues/470 (and the [meeting notes here](https://github.com/npm/rfcs/blob/main/meetings/2021-10-06.md)) … I'm a little unclear on the status of what's left to do here. I'm willing to help if I can!

There were a few action items created as part of that meeting:

> **Actions:**
>   - [x] land npm-packlist in cli (@wraithgar)
>   - [ ] RFC about running prepare scripts for linked bundled deps
>   - [ ] RFC about workspace layout (ljharb)

Taking the second action item first, is that something I should take on? I think it might be blocked by the first item on the list because right now the suggested workaround (below) isn't working with npm@8 or npm@7 … yet?

> ```json
> {
>   "name": "lambda-a",
>   "dependencies": {
>     "b": "file:../b"
>   },
>   "bundleDependencies": [
>     "b"
>   ],
>   "scripts": {
>     "prepare": "cd ../b; npm run prepare"
>   }
> }
> ```

Is there any eta on landing npm-packlist in the cli? I've created a demo repo [here](https://github.com/mysterycommand/multi-app-monorepo) to test out the workaround and see if I could come up with a "hacky" implementation of running prepare scripts on local bundled deps, but currently when I run `npm pack --dry-run --json -w apps/app-a` in that repo, I get this output:

```json
[
  {
    "id": "@multi-app-monorepo/app-a@1.0.0",
    "name": "@multi-app-monorepo/app-a",
    "version": "1.0.0",
    "size": 500,
    "unpackedSize": 838,
    "shasum": "4bd38bbbcc89e412082d48ef45849e53a3512f92",
    "integrity": "sha512-+sHhXSMqS8BCZjuirZwAs2rgs1UU2O+mr+nGlBYSQadIWV9nSNA8NAhO6QW2IUYS6XM5TT5+uSQ2k03Y7+FaWg==",
    "filename": "@multi-app-monorepo/app-a-1.0.0.tgz",
    "files": [
      {
        "path": "index.js",
        "size": 133,
        "mode": 420
      },
      {
        "path": "package.json",
        "size": 705,
        "mode": 420
      }
    ],
    "entryCount": 2,
    "bundled": []
  }
]
```

I'd expect the package.json and index.js from pkgs/pkg-c to show up in the packed output of app-a (maybe under that `"bundled"` key?) … indeed I believe this is the point of the npm-packlist action item. Just wanting to make sure I'm not missing something.

**Comment by mysterycommand** [comment, marked as answer]
cc/ @isaacs @ljharb @wraithgar who were at the meeting and part of the conversation and hopefully still have some context

**Reply by MylesBorins** [reply]
The right place to have this conversation is in the RFC repo not in discussions. Please open an issue there.

**Reply by MylesBorins** [reply]
Also, hi Matt!
